### PR TITLE
Added code to close auth transaction in order to prevent unneeded void btns showed in admin panel after order is canceled

### PR DESCRIPTION
**Pull request by KyerdnA**
# Description
Syncing magento & bolt logic when order is canceling or voiding from magento admin panel. 

Fixes: https://app.asana.com/0/1201931884901947/1204190259084679

#changelog Added plugins to close auth transaction in order to prevent unneeded void btns showed in admin panel after order is canceled

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 

- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my ticket link and provided a changelog message.

### CODE REVIEW

**Comment by danac-gs** [comment]
@KyerdnA the same problem exists with capturing as well.

File: Helper/Order.php [hunk 1]
```diff
@@ -1116,6 +1126,17 @@ public function voidTransactionOnBolt($transactionId, $storeId)
             throw new LocalizedException(__('Payment void error.'));
         }
 
+        $payment = $order->getPayment();
+        $transaction = $this->transactionRepository->getByTransactionType(
```
Comment (vitaliyreznikov) [inline Helper/Order.php:1130]: Let's put it into catch throw block: if transaction doesn't exist for some reason it's better for us to proceed.

File: Plugin/Magento/Sales/Controller/Adminhtml/Invoice/CapturePlugin.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     *
+     * @param Capture $subject
+     * @param $result
+     * @return \Magento\Framework\Controller\ResultInterface
+     * @throws \Magento\Framework\Exception\InputException
+     */
+    public function afterExecute(Capture $subject, $result)
+    {
+        $invoice = $this->invoiceRepository->get($subject->getRequest()->getParam('invoice_id'));
+        $order = $invoice->getOrder();
```
Comment (vitaliyreznikov) [inline Plugin/Magento/Sales/Controller/Adminhtml/Invoice/CapturePlugin.php:56]: Can we move this to internal bolt code as well?

File: Helper/Order.php [hunk 3]
```diff
@@ -2379,6 +2389,19 @@ public function updateOrderPayment($order, $transaction = null, $reference = nul
 
         $payment_transaction->save();
 
+        if ($transactionType == Transaction::TYPE_VOID || ($transactionType == Transaction::TYPE_CAPTURE && $this->isFullyCaptured($order, $transaction))) {
+            $orderTransaction = $this->transactionRepository->getByTransactionType(
```
Comment (vitaliyreznikov) [inline Helper/Order.php:2393]: Do I understand correctly that `getByTransactionType` never throws an exception?

**Review by vitaliyreznikov — APPROVED** [review]

**Review by vitaliyreznikov — APPROVED** [review]

File: Helper/Order.php (hunk 3 above)
Comment (KyerdnA) [reply Helper/Order.php:2393]: It does. InputException
I was going to add it before merging.

**Pull request merged**